src/dbnode/storage/bootstrap/process.go

```diff
@@ -236,6 +242,23 @@ func (b bootstrapProcess) Run(
 		namespacesRunFirst,
 		namespacesRunSecond,
 	} {
+		for _, entry := range namespaces.Namespaces.Iter() {
+			ns := entry.Value()
+			// Check if snapshot-type ranges have advanced while bootstrapping previous ranges.
+			// If yes, return an error to force a retry
+			if persistConf := ns.DataRunOptions.RunOptions.PersistConfig(); persistConf.Enabled &&
+				persistConf.FileSetType == persist.FileSetSnapshotType {
+				upToDateDataRanges := b.targetRangesForData(b.nowFn(),
+					ns.Metadata.Options().RetentionOptions())
```

Comment:
This would never catch up if someone is running with crazy short block sizes. Not sure if this is a practical problem to worry about, though.

Reply:
Well, if the block size is small, there should be less data to bootstrap (unless the block size was made small due to counter bigger than usual amounts of data coming in)